File: Uru_NaturalBooking/Model/ForRequest/ReserveModelForRequest.cs
using Domain;
using System;
using System.Collections.Generic;
using System.Text;

namespace Model.ForRequest
{
    public class ReserveModelForRequest : ModelBaseForRequest<Reserve, ReserveModelForRequest>
    {
        public string Name { get; set; }

        public string LastName { get; set; }

        public string Email { get; set; }

        public DateTime CheckIn { get; set; }

        public DateTime CheckOut { get; set; }

        public int QuantityOfAdult { get; set; }

        public int QuantityOfChild { get; set; }

        public int QuantityOfBaby { get; set; }

        public Guid IdOfLodgingToReserve { get; set; }

        public ReserveModelForRequest() { }

        public override Reserve ToEntity() => new Reserve()
        {
            Name = Name,
            LastName = LastName,
            Email = Email,
            CheckIn = CheckIn,
            CheckOut = CheckOut,
            QuantityOfAdult = QuantityOfAdult,
            QuantityOfChild = QuantityOfChild,
            QuantityOfBaby = QuantityOfBaby
        }; 
    }
}

File: Uru_NaturalBookingFrontend/Uru_NaturalBookingFrontend/src/app/models/ReserveModelForRequest.ts
export class ReserveModelForRequest {
  public name: string;

  public lastName: string;

  public email: string;

  public checkIn: Date;

  public checkOut: Date;

  public quantityOfAdult: number;

  public quantityOfChild: number;

  public quantityOfRetired: number;

  public quantityOfBaby: number;

  public idOfLodgingToReserve: string;

  public constructor(init?: Partial<ReserveModelForRequest>) {
    Object.assign(this, init);
  }
}
